erts/emulator/nifs/common/prim_file_nif.c

```diff
@@ -1399,3 +1404,81 @@ static ERL_NIF_TERM altname_nif(ErlNifEnv *env, int argc, const ERL_NIF_TERM arg
 
     return enif_make_tuple2(env, am_ok, result);
 }
+
+static ERL_NIF_TERM copy_file_range_available_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]) {
+    off64_t ret;
+
+    ret = syscall(SYS_copy_file_range, -1, NULL, -1, NULL, 0, 0);
+
+    if (ret == -1 && errno == ENOSYS) {
+        return enif_make_atom(env, "false");
+    } else {
+        return enif_make_atom(env, "true");
+    }
+}
+
+static ERL_NIF_TERM copy_file_range_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]) {
```

Comment:
changed to `copy_file_nif`

Reply:
Just a comment, RabbitMQ (for example) could greatly benefit from a copy_file_range API that didn't copy the entire file. :)

Reply:
Then let's add both, keeping the specifics (fallback etc) in the platform-specific parts. :-)

Reply:
@jhogberg Should we treat the third argument (length) as optional in the existing function, or should we create a separate function called for example `copy_part_file/3`? In Windows, we utilize `CopyFileW(3)`, but for partial copying, It will require a different approach.

Reply:
I'd prefer a separate function, and think `copy_range` would be a good name since `file` is a bit redundant given the module name (`file:copy_part_file` ...).

Reply:
@jhogberg I have implemented a new function called `copy_range/3`. Additionally, I've added `efile_copy_range_int/3`, which contains code utilized by both `efile_copy_file/2` and `efile_copy_range/3`. I haven't found any similar internal shared functions in the primary file. Therefore, if this approach is incorrect, I can redo it without using the internal function. Implemented `copy_range/3` for Windows API, but not sure if implementation is correct, tried to do in same style al other functions in `win_prim_file.c`. Also will recheck for error handling, have a feeling that I forget to check something.